docs/doc_templates/lorenz-example.md

```diff
@@ -0,0 +1,216 @@
+given these starting coordinates.
+
+Fortunately, Kotlin supports operator overloading, so this will look exactly like a normal
+program even though you're dealing with objects representing probability distributions.
+
+```kotlin
+{% snippet LorenzTimestep %}
+```
+
+Let's define a function that can compute the 10 timesteps in a window using the function you defined above. 
+
+```kotlin
+{% snippet LorenzWindow %}
+```
+ 
+You can now apply the observations of the X coordinate from the "Real World" system to each timestep
+in the window calculated above. 
+
+Let's define a function that does this.
+
+```kotlin
+{% snippet LorenzObservations %}
+```
+
+This function is iterating through each of the timesteps you calculated above, extracting the Gaussian
+distribution that describes the X coordinate, and observing it to be the value taken from the
+"Real World" system.
+
+Brilliant, now let's calculate the most likely value for the X, Y and Z coordinates.
+
+Hopefully by now the code for performing Gradient Optimisation will look familiar to you:
+
+```kotlin
+{% snippet LorenzOptimise %}
+```
+
+Now you've calculated the most probable values for the graph, you need to feed these into the next window.
+As you're performing 10 timesteps in a window, you want to take the value of the 10th timestep as the
+starting point of the next window.
+
+Let's define a function to do that:
+
+```kotlin
+{% snippet LorenzGetWindowVals %}
+```
+
+Great, now you need to feed these back into the next window.
+
+If you'll recall, at the start of the window, you defined the initial coordinates of the graph as:
+
+```kotlin
+{% snippet LorenzStartValues %}
+```
+
+Therefore, let's recalculate these Gaussians to include the calculated value of the most probable coordinates:
+
+```kotlin
+{% snippet LorenzNewStartValues %}
+```
+
+## Results
+
+Can you have a guess at what you think the results will be before you run the code?
+
+We are performing 10 timesteps in each window: how many windows do you think it will take before the
+error between the probabilistic model and the "Real World" model is negligible?
+
+Printing out the error of each window produces the following results.
+
+```
+Error: 4.528976372510882
+Error: 3.72098313476314
+Error: 2.665636788694244
+Error: 1.812040602844943
+Error: 1.2950274565457909
+Error: 1.040466635607921
+Error: 0.7338159078803722
+Error: 0.38109769602303945
+Error: 0.1774717811757308
+Error: 0.11441697844622131
+Error: 0.0918083820220631
+Error: 0.06013662151356879
+Error: 0.030568642144244644
+Error: 0.02316773884356288
+Error: 0.015850612891648755
+Error: 0.008663664263278796
+```
+
+It takes 16 windows before the probabilistic model has converged on the "Real World" system, from only an
+observation of the X coordinate!
+
+## Code
+
+Here is the completed code if you'd like to run it yourself.
+
+```kotlin
+{% snippet LorenzFull %}
+```
+
+## Java Code
+
+Here is the same implementation, but done in Java.
```

Comment:
we have a python version now too - could add that here

Reply:
I think we should wait until we add a Python section to the docs before we start dropping little snippets